### PR TITLE
Builds on Win32

### PR DESCRIPTION
**Pull request by ooeygui**

### CODE REVIEW

File: kuka_eki_hw_interface/include/kuka_eki_hw_interface/kuka_eki_hw_interface.h [hunk 1]
```diff
@@ -38,10 +38,10 @@
 #include <vector>
 #include <string>
 
-#include <boost/asio.hpp>
-
 #include <ros/ros.h>
 #include <controller_manager/controller_manager.h>
+
+#include <boost/asio.hpp>
```
Comment (gavanderhoorn) [inline kuka_eki_hw_interface/include/kuka_eki_hw_interface/kuka_eki_hw_interface.h:44]: Is this change needed to work-around `boost/asio.hpp` including `Windows.h` somewhere and then conflicting with the `ERROR` `enum class` in `ros_control`'s `robot_hw.h` (or something similar)?

It appears we're running into *something* which causes problems with the `ERROR` `#define` in https://github.com/ros-industrial/abb_robot_driver/issues/17.

Comment (ooeygui) [reply kuka_eki_hw_interface/include/kuka_eki_hw_interface/kuka_eki_hw_interface.h:44]: Yes, I can add a comment for that.
It's really unfortunate that #defines are not namespaced, and that 30 years ago someone thought #define ERROR was a good idea.

Comment (gavanderhoorn) [reply kuka_eki_hw_interface/include/kuka_eki_hw_interface/kuka_eki_hw_interface.h:44]: Yes, that's what I was thinking as well :)

The strange thing is: some users of `abb_robot_driver` report this problem with ROS Melodic and ROS Noetic, while the main developer (@jontje) doesn't run into it (on Melodic).

File: kuka_rsi_hw_interface/include/kuka_rsi_hw_interface/udp_server.h [hunk 2]
```diff
@@ -44,6 +44,18 @@
 #include <string>
 #include <stdexcept>
 
+#ifdef WIN32
+#include <windows.h>
+#undef ERROR
+#undef OK
```
Comment (gavanderhoorn) [inline kuka_rsi_hw_interface/include/kuka_rsi_hw_interface/udp_server.h:50]: A bit late, but would using what @seanyen did in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1023 perhaps be possible here (and cleaner)?

Comment (ooeygui) [reply kuka_rsi_hw_interface/include/kuka_rsi_hw_interface/udp_server.h:50]: Oh yea, that is a better fix. I'll update it, thanks!
